### ISSUE TITLE
New strategy based in indicators

### ISSUE THREAD

**Issue by thesam1545** [chrisleekr/binance-trading-bot#327]
I have watched a video about an expert trading guy, who talked about the best indicators that should be used to buy/sell in the crypto world:
https://www.youtube.com/watch?v=33lGueGyrc4&t=3s&ab_channel=TheSecretMindset

He claims that there the indicators that should be applied are: VWAP, Heikin-Ashi candles, MACD and Bollinger Bands

This bot is very well made, but I have not made meaningful profits with the Trailing Stop Orders...
I very little coding knowledge, but I would like to see these indicators applied in this bot. Anyone here would like to give them try and try to implement them in this bot?

**Comment by chrisleekr** [comment]
Originally, the bot was developed to use MACD and Bollinger Bands. 
However, those indicators were causing very wrong signals with the live market data.
What I mean is, if you just look at the charts for the past data, it does make sense with MACD/Bollinger Bands.
With feeding the live data, it didn't really work.

Trailing stop orders is just a method to follow the lowest/highest price, not the indicator. 
The buy indicator in this bot is simply, the lowest value within configured time. 
The sell indicator in this bot is simply, a certain amount of the profit percentage.

Now, adding indicators is not an issue. But how to configure/calculate and use the signal are the issue.
Although, I am a bit sceptical because all indicators with the past data always making sense. 
Unless you can predict the future, the indicators will be always behind. 

I had many suggestions that traders want to have more indicators. However, I am not a professional in trading or math at all. 
If you can give me a more specific formula and configurable variables, then I can help to implement them.

**Comment by thesam1545** [comment]
First of all, thank you for you reply and for this great tool you developed and keep updating.

Now, I understand what you said, but what understood from the video is that those 4 indicators should be used combined because they cover different signals... so maybe those indicators missing should be added.. I don't know let me know, your opinion.

Let me think a bit more about this, because it would be great to further create a tool that could buy and sell in a more efficient way.

**Comment by thesam1545** [comment]
Basically, lets take for example XVGEUR pair... 

@chrisleekr in your opinion, if we take in consideration the technical analisys based in indicators, available in TradingView platform: https://br.tradingview.com/symbols/XRPEUR/technicals/ and use the sell and buy signal  in the "Summary" gauge to make decisions on sell/buy and quantity, in the selected time frame, we would be losing money? Is TradingView providing "wrong information"? 

I really don't know, but my curiosity is really high.

**Comment by eth-man** [comment]
I know a very good bot that uses TV signals (not sure if we would mention the company name)
 you can get very high success rates with the combination RSI below 70 and TV, buy, strong buy signals.
I think these are good indicators because of the crowd use itself can raise the price of the coin
it is not always accurate , you'll still have red bags

**Comment by chrisleekr** [comment]
> @chrisleekr in your opinion, if we take in consideration the technical analysis based in indicators, available in TradingView platform: https://br.tradingview.com/symbols/XRPEUR/technicals/ and use the sell and buy signal in the "Summary" gauge to make decisions on sell/buy and quantity, in the selected time frame, we would be losing money? Is TradingView providing "wrong information"?

I never said the technical analysis will lose money. It is more of an issue to code/configure the indicators, which takes a lot of my time and energy to develop the features. If there are easy methods to get those indicators like this TradingView technical page, definitely I can add in.

That trading view technicals page is pretty interesting. 

The best way is if there is a free API service to provide those indicators with actions, we can hook with buy/sell configurations.
The alternative way will be parsing their website (or HTTP call), but this may be illegal so definitely not suggested.
The last option is calculating all indicators in the bot itself, which I want to avoid because it will be complex to add all signals to the bot.

**Comment by thesam1545** [comment]
@chrisleekr @eth-man 
Tradingview only has a payed feature that gives an alert when an indicator crosses a certain value... https://www.youtube.com/watch?v=-wT9h9Nc9sk&ab_channel=PartTimeLarry
But if we follow this logic, we would have the indicator we want,but still had to decide what indicators to use, when to buy and how much based on the indicators value. 
I think this way is a no go, because it requires more work and effort when we already have all the work made for us in the technical analysis webpage. However, TradingView does not have any direct feature or API that shares the when to buy/sell "gauge".

In my opinion, parsing their website (or HTTP call), if it is a relative easy job to do, would be the way. And we could evaluate the bot success from it.

**Comment by chrisleekr** [comment]
I actually found this - https://www.tradingview.com/widget/technical-analysis/

For the first cut, I may be able to add this to the frontend for reference purposes.
This is simply using their provided free service.

![image](https://user-images.githubusercontent.com/5715919/134760715-b4a2fe4b-6d07-42a5-bf8c-7624c2204551.png)

> In my opinion, parsing their website (or HTTP call), if it is a relative easy job to do, would be the way. And we could evaluate the bot success from it.

It's not like just getting the HTML as this must be rendered via their Javascript.
It should load the browser to get the content.

**Comment by thesam1545** [comment]
Nice thing you made, congratulations. @chrisleekr  :)
But we still have to decide how many times the bot will sell or buy during a timelapse. And how much crypto it will sell or  buy based in the gauge position.

My suggestion is that it sells or buy the available asset in a percentage of the gauge position.
Strong sell - 100% asset sell
Strong buy - 100% asset buy
Neutral - do nothing

I think the bot can make it once every minute, or once every second, but I really don't know what to expect.
We should add the restriction to not sell lower than the bought price, just in case anything goes wrong.

**Comment by thesam1545** [comment]
Any news guys? @chrisleekr @eth-man

**Comment by thesam1545** [comment]
> I am not sure this is a good strategy. You buy/sell based on the gauge position, then there is no reason to use this bot because that is a completely different method.

I just though that was a simple way to approach our problem. If we use the widget + look for the lowest curent price in comparison with previous prices (current bot method), the bot will take a long time to buy. And will take a long time to sell too.

**Comment by chrisleekr** [comment]
> I just though that was a simple way to approach our problem. If we use the widget + look for the lowest curent price in comparison with previous prices (current bot method), the bot will take a long time to buy. And will take a long time to sell too.

Yeah, it may take some time, but it will give some safeguard to buy at a high price.
I am thinking to do it in this way in addition to the one method above-mentioned.

- Prevent buy when summary->recommendation is Sell/Strong Sell
- Reschedule auto-trigger buy when the recommendation is Neutral/Sell/Strong Sell

**Comment by thesam1545** [comment]
Looks a good logic to test and see @chrisleekr .

Is it possible for me to test this too?

**Comment by chrisleekr** [comment]
It's still under development. Still need to write test cases as well. It will take some time.

**Comment by thesam1545** [comment]
> It's still under development. Still need to write test cases as well. It will take some time.

Ok, any doubt or advice or help (if I have the know how) I can give tell me. :)

**Comment by eth-man** [comment]
@chrisleekr , that's looks wonderful 
Ii looks like you using 15 minutes interval 
Will it be possible to allow the user change it ?
Or chose multiple intervals?l for same coin ?

Thank you

**Comment by chrisleekr** [comment]
@eth-man Of course. It will use candle intervals. I am preparing the PR now.

**Comment by eth-man** [comment]
Thanks🙏

**Comment by thesam1545** [comment]
> @eth-man Of course. It will use candle intervals. I am preparing the PR now.

Congratulations and thanks!
